### PR TITLE
Feat: Address Management On Live

### PR DESCRIPTION
**Pull request by Rishi05051997**
Peer Members, please review Address Management 
- Add Address
- Remove Address
- Edit Address 
- Display Existing Stored Address

https://user-images.githubusercontent.com/68817653/161065231-e01e6068-b702-4bc9-b336-6819f796e6e0.mp4

### CODE REVIEW

File: src/components/Cart/Address/AddressListing.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                        <div key={_id}>
+                            <div className="head-3 bold highlightMainText mar-y-2">{type}</div>
+                            <div className="displayFlexRow address-mgmt">
+                                <div className="text-2 displayFlexColumn">
+                                    <div className="bold">{name}</div>
+                                    <div className="text-2 secondory-text">{mobileNo}</div>
+                                    <div className="text-2 secondory-text">{houseNo} {city}</div>
+                                    <div className="text-2 secondory-text">Unkown {district}</div>
+                                    <div className="text-2 secondory-text">{state}-{zipCode}</div>
+                                    <div className="text-2 secondory-text">{country}</div>
```
Comment (viru2001) [inline src/components/Cart/Address/AddressListing.jsx:25]: why you are using div here? you can use p tag also

File: src/services/address-service.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+// import axios from "axios";
+
+// const encodedToken = localStorage.getItem("token");
+
+const address = [
+    {
+        _id: Math.floor((Math.random() * 100) + 1),
```
Comment (viru2001) [inline src/services/address-service.js:7]: you can use uuid for getting random id's

**Review by viru2001 — APPROVED** [review]
- Code is good
- I liked how you used destructuring
- UI is clean

just gave small suggestions on which you can work

File: src/components/Cart/Address/AddressListing.jsx (hunk 1 above)
Comment (Rishi05051997) [reply src/components/Cart/Address/AddressListing.jsx:25]: as you know there are some internal properties come with p, h1 tag so for better I used div here

File: src/services/address-service.js (hunk 2 above)
Comment (Rishi05051997) [reply src/services/address-service.js:7]: Ok

File: src/components/Cart/Address/add-new-address.jsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            setErrorMsg((msg) => ({ ...msg, mobileNo: "Please Enter a Valid Mobile No" }));
+            errorStatus = false;
+        }
+        if (!/^[1-9][0-9]{5}$/.test(address.zipCode)) {
+            setErrorMsg((msg) => ({ ...msg, zipCode: "Please Enter a Valid Zip Code" }));
+            errorStatus = false;
+        }
+
+        return errorStatus
+    }
```
Comment (pratik-kumar-sahu) [inline src/components/Cart/Address/add-new-address.jsx:36]: - this is a type of common method you can move these type of functions in other helper files to make code more clean.

**Review by pratik-kumar-sahu — APPROVED** [review]
- Overall code and ui looks good. Nice work
- State Handling is also good.

File: src/components/Cart/Address/add-new-address.jsx (hunk 3 above)
Comment (Rishi05051997) [reply src/components/Cart/Address/add-new-address.jsx:36]: Yes as a refactoring I will do this one.

**Pull request merged**